### PR TITLE
Only asynchronous function call with new wait feature

### PR DESCRIPTION
**Pull request by andeplane**

### CODE REVIEW

File: cognite/experimental/data_classes/functions.py [hunk 1]
```diff
@@ -145,6 +146,18 @@ def __init__(
     def get_logs(self):
         return self._cognite_client.functions.calls.get_logs(call_id=self.id, function_id=self._function_id)
 
+    def update(self):
+        latest = self._cognite_client.functions.calls.retrieve(id=self.id)
```
[the remark this answers is not in the document]
Comment (andeplane) [reply cognite/experimental/data_classes/functions.py:150]: I was thinking about that, but I think it is easier to read like it is now :D

File: tests/tests_unit/test_api/test_functions.py [hunk 2]
```diff
@@ -92,23 +92,20 @@ def mock_functions_delete_response(rsps):
 
 @pytest.fixture
 def mock_functions_call_completed_response(rsps):
-    response_body_async = BASE_CALL.copy()
     response_body_sync = BASE_CALL.copy()
-    response_body_sync["response"] = "Hello World!"
+    response_body_sync["response"] = {"pi": 3.1415}
```
[the remark this answers is not in the document]
Comment (andeplane) [reply tests/tests_unit/test_api/test_functions.py:96]: I agree!

File: tests/tests_unit/test_api/test_functions.py [hunk 3]
```diff
@@ -92,23 +92,20 @@ def mock_functions_delete_response(rsps):
 
 @pytest.fixture
 def mock_functions_call_completed_response(rsps):
-    response_body_async = BASE_CALL.copy()
     response_body_sync = BASE_CALL.copy()
```
[the remark this answers is not in the document]
Comment (andeplane) [reply tests/tests_unit/test_api/test_functions.py:95]: Yep

File: cognite/experimental/data_classes/functions.py [hunk 4]
```diff
@@ -116,7 +117,7 @@ class FunctionCall(CogniteResource):
         id (int): A server-generated ID for the object.
         start_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
         end_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
-        response (str): Response from the function. The function must return a JSON serializable object.
+        response (dict): Response from the function. The function must return a dictionary.
```
Comment (andeplane) [inline cognite/experimental/data_classes/functions.py:120]:
```suggestion
        response (dict): Response from the function. The function must return nothing, or a JSON serializable dictionary.
```

Comment (HMEiding) [reply cognite/experimental/data_classes/functions.py:120]: In view of the discussion with Dag earlier today, I think we should decide on the response schema before merging this.

Comment (HMEiding) [reply cognite/experimental/data_classes/functions.py:120]: As the API is today, `dict` is not necessary, just that it's json serializable.

File: pyproject.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "cognite-sdk-experimental"
-version = "0.7.2"
+version = "0.7.3"
```
Comment (andeplane) [inline pyproject.toml:3]:
```suggestion
version = "0.8.0"
```

File: cognite/experimental/data_classes/functions.py [hunk 6]
```diff
@@ -145,6 +146,18 @@ def __init__(
     def get_logs(self):
         return self._cognite_client.functions.calls.get_logs(call_id=self.id, function_id=self._function_id)
 
+    def update(self):
```
Comment (andeplane) [inline cognite/experimental/data_classes/functions.py:149]: Should we document this too?

**Pull request merged**